scripts/setupPythonEnvironment.bash

```diff
@@ -144,31 +120,53 @@ then
     fi
 
     # Setup the virtual environment
+    if $VERBOSE
+    then
+        echo "Virtual environment path:"
+        echo $VIRTUAL_PATH
+    fi
+
     mkdir -p $VIRTUAL_PATH/$VIRTUAL_NAME
     $PYTHON_TARGET -m virtualenv --system-site-packages $VIRTUAL_PATH/$VIRTUAL_NAME
 
-    # Install packages
-    echo "Installing packages..."
     PYTHON_TARGET=$VIRTUAL_PATH/$VIRTUAL_NAME/bin/python
```

Comment:
It looks like we're swinging between the `python3` of the `PATH`, the `Python3_ROOT_DIR` if defined, and a `virtualenv` created on the fly.

Could we ask the use to explicitly define `Python3_ROOT_DIR` and let him pass this information to the script as a parameter? (If not provided, we die with an explicit message)
That way, he can use whatever env system he wants and we get rid of any troubleshooting on our side? (Surely logging which python is being used can help).

Reply:
E.g. I've added
```shell
if [[ "${RUN_INTEGRATED_TESTS}" = true ]]; then
  echo "Running the integrated tests has been requested."
  # We install the python environment required by ATS to run the integrated tests.
  or_die apt-get install -y virtualenv python3-dev python-is-python3
  ATS_PYTHON_HOME=/tmp/run_integrated_tests_virtualenv
  or_die virtualenv ${ATS_PYTHON_HOME}
  ATS_CMAKE_ARGS="-DATS_ARGUMENTS=\"--machine openmpi --ats openmpi_mpirun=/usr/bin/mpirun --ats openmpi_args=--allow-run-as-root --ats openmpi_procspernode=2 --ats openmpi_maxprocs=2\" -DPython3_ROOT_DIR=${ATS_PYTHON_HOME}"
fi
```
in the ci file [here](https://github.com/GEOS-DEV/GEOS/pull/2801/files#diff-0cef284cfdb08fea23f36d5d918fe37db5c5af609b2837dc27253a354dc1e641). (see `DPython3_ROOT_DIR` at the end)

Reply:
> It looks like we're swinging between the `python3` of the `PATH`, the `Python3_ROOT_DIR` if defined, and a `virtualenv` created on the fly.
> 
> Could we ask the user to explicitly define `Python3_ROOT_DIR` and let him pass this information to the script as a parameter? (If not provided, we die with an explicit message) That way, he can use whatever env system he wants and we get rid of any troubleshooting on our side? (Surely logging which python is being used can help).

That's fair.  If you want, we could remove the default (from the path) and the virtualenv construction method.  These are all hold-overs from historical development/testing, and are rarely used now.

Reply:
I guess that the script would be simplified into: give me an environment in which I can install my python packages.
If I can't, here's the error message, deal with it and start again.